bundles/org.eclipse.e4.ui.model.workbench/src/org/eclipse/e4/ui/model/XPathHelper.java

```diff
@@ -0,0 +1,36 @@
+/*******************************************************************************
+ * Copyright (c) 2025, 2025 Hannes Wellmann and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Hannes Wellmann - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.e4.ui.model;
+
+import java.util.stream.Stream;
+import org.eclipse.core.runtime.ILog;
+import org.eclipse.e4.emf.xpath.XPathContextFactory;
+import org.eclipse.emf.ecore.EObject;
+
+public class XPathHelper {
+
+	@SuppressWarnings("removal")
+	public static <T> Stream<T> findMatchingElements(EObject contextBean, String xpath, Class<T> clazz) {
```

Comment:
Does this needs to be public API or can we move it to an internal package?

Reply:
The package is already internal and just a friend to `org.eclipse.e4.ui.workbench`:
`org.eclipse.e4.ui.model;x-friends:="org.eclipse.e4.ui.workbench,org.eclipse.e4.ui.workbench.swt,org.eclipse.ui.workbench"`